app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { IManagedArtifactVersion } from '../domain';
+import { useEventListener } from '../presentation';
+
+import { ArtifactDetailHeader } from './ArtifactDetailHeader';
+
+import './ArtifactDetail.less';
+
+export interface IArtifactDetailProps {
+  name: string;
+  version: IManagedArtifactVersion;
+  onRequestClose: () => any;
+}
+
+export const ArtifactDetail = ({ name, version, onRequestClose }: IArtifactDetailProps) => {
+  const keydownCallback = ({ keyCode }: KeyboardEvent) => {
+    if (keyCode === 27 /* esc */) {
+      onRequestClose();
+    }
+  };
+  useEventListener(document, 'keydown', keydownCallback);
```

Comment:
it seemed nice to exit out of this on escape. later I might even add previous/next keyboard shortcuts for versions.

Reply:
j, k, up, down? 😅

Reply:
_literally_ what i was thinking of. too many 2000s-era flash games and Gmail threads.